### ISSUE TITLE
"dev as stable" must be handled as stable

### ISSUE THREAD

**Issue by mvorisek** [composer/composer#11582]
```
{
    "require": {
        "doctrine/dbal": "~3.4.5 || ~3.5.1 || ~3.6.0 || 4.0.x-dev as 4.0.0"
    },
    "prefer-stable": true
}
```
must install `4.0.x` as aliased as `4.0.0` stable

currently `"prefer-stable": false` is needed

**Comment by fredden** [comment]
I disagree with this feature request. `dev-whatever` and `4.0.x-dev` are *not* stable. Using an alias of "as 4.0.0" just means that a particular version is allowed to be used as a substitute for when another package wants something that matches the alias; it doesn't change the stability of the actual version being installed.
For example, using a constraint of `1.2.3 as 2.0.1` means that if something wants `^2.0` that'll be satisfied by version `1.2.3` which is installed. If you want to have non-stable packages installed, then you'll need to tell Composer that you want to have non-stable packages installed. See https://getcomposer.org/doc/articles/versions.md#stabilities and https://getcomposer.org/doc/04-schema.md#minimum-stability.

**Comment by Seldaek** [comment]
prefer-stable is correct here in preferring the more stable version. The alias makes the package have both versions, but it is not a way to replace the version entirely, nor its stability.
